token.go

```diff
@@ -14,12 +14,14 @@ type Keyfunc func(*Token) (interface{}, error)
 // Token represents a JWT Token.  Different fields will be used depending on
 // whether you're creating or parsing/verifying a token.
 type Token struct {
-	Raw       string                 // Raw contains the raw token.  Populated when you [Parse] a token
-	Method    SigningMethod          // Method is the signing method used or to be used
-	Header    map[string]interface{} // Header is the first segment of the token in decoded form
-	Claims    Claims                 // Claims is the second segment of the token in decoded form
-	Signature []byte                 // Signature is the third segment of the token in decoded form.  Populated when you Parse a token
-	Valid     bool                   // Valid specifies if the token is valid.  Populated when you Parse/Verify a token
+	Raw           string                 // Raw contains the raw token.  Populated when you [Parse] a token
+	Method        SigningMethod          // Method is the signing method used or to be used
+	Header        map[string]interface{} // Header is the first segment of the token in decoded form
+	Claims        Claims                 // Claims is the second segment of the token in decoded form
+	Signature     []byte                 // Signature is the third segment of the token in decoded form.  Populated when you Parse a token
+	Valid         bool                   // Valid specifies if the token is valid.  Populated when you Parse/Verify a token
+	jsonEncoder   JSONEncoder            // jsonEncoder is the custom json encoder/decoder
```

Comment:
I know that the `Parser` has these fields directly embedded into the struct but in another PR we are trying to extract all the options into an unexported config options struct to not clutter the "main" struct. I would suggest doing this from the start and to move jsonEncoder and base64Encoder into a separate private config options struct.